src/App.js

```diff
@@ -1,55 +1,102 @@
 import React, { Component } from 'react'
 import { Router, Route } from 'react-static'
-
-import LanguageWrapper from "js/components/LanguageWrapper"
+import Routes from 'react-static-routes'
+import { IntlProvider } from 'react-intl'
+import locale from 'browser-locale'
+import Cookies from 'js-cookie'
+import ReactGA from 'react-ga'
+import GoogleAnalyticsPageView from 'js/helpers/GoogleAnalyticsPageView'
 
 // page_sections
-import I18nBanner from "js/page_sections/I18nBanner"
+import LanguageSelectBanner from "js/page_sections/LanguageSelectBanner"
 import Header from "js/page_sections/Header"
 import Footer from "js/page_sections/Footer"
 
-// pages
-import Home from "js/pages/Home"
-import Search from "js/pages/Search"
-import Services from "js/pages/Services"
-import Service from "js/pages/Service"
-import Topic from "js/pages/Topic"
-import Department from "js/pages/Department"
 
-import Routes from 'react-static-routes'
+import { SUPPORTED_LANGUAGES } from 'js/constants/languages'
+
 
 
 import 'css/coa.css'
 
+ReactGA.initialize('UA-110716917-2', { debug: true });
+
+
 class App extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      lang: this.setLanguage()
+    }
+    this.daysUntilCookieExpires = 10 * 365;
+  }
+
+  parseBrowserLanguageCode = () => {
+    // Normally, we only want the two letter lowercased language abbreviation
+    // bc we aren't worried about locale (ex: en-US vs. en-UK) at this point,
+    // just language.
+    if (typeof document !== 'undefined') {
+      const twoLetterLangCode = locale().split('-')[0].toLowerCase();
+      // But we do want to support two types of Chinese (zh-tw & zh-cn)
+      const isChinese = twoLetterLangCode === 'zh';
+      return isChinese ? locale().toLowerCase() : twoLetterLangCode;
+    }
+  }
+
+  setLanguage = () => {
+    const cookieLanguage = Cookies.get('lang');
+    const browserLocale = this.parseBrowserLanguageCode();
+    let language = '';
+
+    const isLanguageCodeInPath = SUPPORTED_LANGUAGES
```

Comment:
I don't think this chunk which pulls the language from the url path still works as is. I don't think we define this.props.urlPathLanguage any longer. This was something we were passing into LanguageWrapper from props.match.params.lang, correct?

Reply:
This may be completely obvious, but I'm a total noob. I'm curious why you opted to pull the language specific logic out of a component and into App.js. I thought the abstraction kept the App.js code super clean.

Reply:
I think it still works, the language is stored in the component state and gets passed into the `IntlProvider` as the locale string. 

I just opted to pull it in because it was the most straightforward path to getting things to work. I'm open to moving it back into its own wrapper, but the App.js didn't seem overly complicated to me as it is.

Reply:
![feb-06-2018 15-35-11](https://user-images.githubusercontent.com/5697474/35885513-ae96a112-0b53-11e8-8c38-b4ce6ce3635c.gif)

Reply:
Hey, I think I wasn't being clear. So if I navigate to a page with a language code in the url, It doesn't seem like the app recognizes this language code. So in the [video](https://drive.google.com/open?id=1TOxOlQ5Pkd1q4XSpCLSmYuHGiDDznFGz), I go to /es/ and you'll see that urlPathLanguage is undefined and english is the selected language in the menu and cookie instead of spanish.

Reply:
yup, you're right. working on fixing it now. Thanks for attaching the video